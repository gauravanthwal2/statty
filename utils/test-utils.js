import { configureStore } from "@reduxjs/toolkit";
import { Provider } from "react-redux";
import { rootReducer } from "../redux/store";
import { render as rtlRender } from "@testing-library/react";

export function renderWithProvider(
  ui,
  {
    preloadedState = {},
    // Automatically create a store instance if no store was passed in
    store = configureStore({ reducer: rootReducer, preloadedState }),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>;
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

export * from "@testing-library/react";

