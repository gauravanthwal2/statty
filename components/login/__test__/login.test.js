import { fireEvent, cleanup, render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import Login from "../../../pages/login";
import OTP from "../OtpForm";
import { renderWithProvider } from "../../../utils/test-utils";

describe("Test the login component", () => {
  afterEach(() => cleanup());
  test("testing", () => {
    // const { getByLabelText} =  render(<Provider store={store}><Login /></Provider>)
    const { getByRole, getByLabelText } = renderWithProvider(<Login />);
    expect(getByRole("button", { name: "Send OTP" })).toBeInTheDocument();
    screen.debug();
  });

  test("Test login Email Input", () => {
    const { getByRole, getByLabelText } = renderWithProvider(<Login />);
    const emailText = getByRole("textbox");
    // userEvent.type(emailText, "abc");
    fireEvent.change(emailText, { target: { value: "abc" } });
    // expect(emailText.value).not.toEqual("abc@publicissapient.com")
    expect(emailText.value).not.toMatch("abc@publicissapient.com");
  });

  test("Test login Blank Email Input", () => {
    const { getByRole, getByTestId, getByLabelText } = renderWithProvider(
      <Login />
    );
    // const emailBtn = getByTestId("loginEmail");
    const emailText = getByRole("textbox");
    fireEvent.click(emailText, { target: { value: "" } });
    expect(emailText.value).not.toEqual("Please enter valid otp");
  });

  test("should show otp input", () => {
    const { getByRole, getByTestId, getByLabelText } = renderWithProvider(
      <OTP />
    );
    const otpBtn = getByTestId("primary");
    const otpNumber = getByRole("textbox");
    userEvent.click(otpBtn, "123");
    expect(otpNumber.value).not.toBeNull();
    // fireEvent.click(otpBTN, {target: {value: ""}})
    // expect(otpNumber.value).not.toEqual("1234")
  });

  test("Test submit OTP Input", () => {
    const {
      getByRole,
      getByTestId,
      getByText,
      getByPlaceholderText,
      getByLabelText,
    } = renderWithProvider(<OTP />);
    const otpBtn = getByTestId("primary");
    const otpNumber = getByRole("textbox");
    // userEvent.type(otpBtnn);
    fireEvent.change(otpNumber, { target: { value: "1234" } });
    userEvent.click(otpBtn);
    expect(otpNumber).toHaveValue("1234");
  });
  test("renders login", () => {
    const { container } = renderWithProvider(<Login />);
    expect(container).toMatchSnapshot();
  });
});
