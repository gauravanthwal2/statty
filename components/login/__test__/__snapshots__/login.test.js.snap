// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the login component renders login 1`] = `
<div>
  <div
    class="backGrnd"
  >
    <div
      class="login"
    >
      <div
        class="loginForm"
      >
        <div
          class="formLable"
        >
          <form>
            <div
              class="row"
            >
              <label
                class="rowChng"
<<<<<<< HEAD
                for="email" >
=======
                for="email"
              >
>>>>>>> 22f94f16ba1468ebc5873269e1fcfdbd56e071af
                Email Address 
                 
                <span
                  class="required"
                >
                  *
                </span>
              </label>
              <input
                class="form-control"
                id="email"
                name="email"
                required=""
                type="email"
                value=""
              />
              <p
                class="example"
              >
                Example: Smigel@publicissapient.com
              </p>
            </div>
            <div
              class="row justify-content-center"
            >
              <button
                class="primaryButton"
                data-testid="primary"
                type="submit"
              >
                Send OTP
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
